mir_eval/chord.py

```diff
@@ -326,19 +328,17 @@ def validate_chord_label(chord_label):
         Chord label to validate.
 
     """
-    # Test for single special characters
-    for one_char in [':', '/', '(', ')']:
-        if chord_label.count(one_char) > 1:
-            raise InvalidChordException(
-                "Chord label may only contain one '%s'. "
-                "Received: '%s'" % (one_char, chord_label))
 
-    # Test for closed parens
-    parens = [paren in chord_label for paren in ['(', ')']]
-    if any(parens) and not all(parens):
-        raise InvalidChordException(
-            "Chord label must have closed parentheses. "
-            "Received: '%s'" % chord_label)
+    # This monster regexp is pulled from the JAMS chord namespace,
+    # which is in turn derived from the context-free grammar of
+    # Harte et al., 2005.
+
+    pattern = re.compile(r'''^((N|X)|(([A-G](b*|#*))((:(maj|min|dim|aug|1|5|sus2|sus4|maj6|min6|7|maj7|min7|dim7|hdim7|minmaj7|aug7|9|maj9|min9|11|maj11|min11|13|maj13|min13)(\((\*?((b*|#*)([1-9]|1[0-3]?))(,\*?((b*|#*)([1-9]|1[0-3]?)))*)\))?)|(:\((\*?((b*|#*)([1-9]|1[0-3]?))(,\*?((b*|#*)([1-9]|1[0-3]?)))*)\)))?((/((b*|#*)([1-9]|1[0-3]?)))?)?))$''')  # nopep8
```

Comment:
> I dunno, wcgw?

I generally dislike relying on invisible string concatenation because it's brittle.  If, somehow, the string no longer exists inside parentheses, say because you pull it out to its own variable name `exp = '....'` then it fails silently.  I've been burned by this kind of thing before, so I try to avoid it.

> Isn't this true of literally every string in Python?

Not strings that could have apostrophes in them (which we don't in this case) or could be interpolated via %/{} (which we also don't).  Maybe it's just my own quirk to use `r''' '''` for all regexps just in case those things do pop up naturally in the expression.